### PR TITLE
Adding joule heating source term

### LINKED ISSUE #8220: Add heat source term representing Joule Heating
### Description of the enhancement or error report
Add Kernel & Material for implementing electrical conductivity and joule heating. 

### Rationale for the enhancement or information for reproducing the error
Capability to couple electrical field with existing modules. 

### Identified impact
Additional capability.

### PR DESCRIPTION
**Pull request by SudiptaBiswas**
- Adds kernel for joule heating source term and electrical conductivity material
- closes #8220

### CODE REVIEW

File: modules/heat_conduction/src/kernels/JouleHeatingSource.C [hunk 1]
```diff
@@ -0,0 +1,53 @@
+#include "JouleHeatingSource.h"
+
+template<>
+InputParameters validParams<JouleHeatingSource>()
+{
+  InputParameters params = validParams<HeatSource>();
+  params.addCoupledVar("elec", "Electric potential for joule heating.");
+  params.addCoupledVar("args", "Vector of arguments of the diffusivity");
+  params.addParam<MaterialPropertyName>("electrical_conductivity", "electrical_conductivity", "material property providing electrical resistivity of the material.");
```
Comment (dschwen) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: Conductivity is the reciprocal of resistivity.

Comment (dschwen) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: I suggest cleaning up the mismatch of the terms and maybe only calculate one of the two quantities.

Comment (SudiptaBiswas) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: Calculating electrical conductivity in the material helps in using MatDiffusion kernel directly without parsed functions for the electric potential variable. May be I can skip storing resistivity as material property.

Comment (SudiptaBiswas) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: BTW how do I activate the tests?

Comment (dschwen) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: You don't, I can. Maybe @permcody can add you to the collaborators group for which tests are run automatically.

Comment (SudiptaBiswas) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: okay.

**Comment by permcody** [comment]
Please ask @brianmoose to add you.

**Comment by brianmoose** [comment]
The invitation has been sent.

**Comment by SudiptaBiswas** [comment]
Thanks @brianmoose.

File: modules/heat_conduction/src/kernels/JouleHeatingSource.C (hunk 1 above)
Comment (dschwen) [inline modules/heat_conduction/src/kernels/JouleHeatingSource.C]: Still needs to say _... providing electrical **conductivity** of the material._

**Review by dschwen — APPROVED** [review]

**Comment by brianmoose** [comment]
To enable automatic testing you can either accept the invitation that was sent to you by email or go to https://github.com/idaholab and accept it there.

**Comment by permcody** [comment]
I added this kernel years ago:
https://github.com/idaholab/moose/blob/devel/modules/misc/src/JouleHeating.C.  It's in the "misc" module and probably hasn't been used since I used it in a toy project. Now that this is going in, we should probably remove the old kernel. I'll open a separate ticket.

**Comment by SudiptaBiswas** [comment]
I probably missed that while developing this. Thanks for pointing out.

**Comment by permcody** [comment]
I can't expect you to dig through everything, every time you need to build
something. It's in a weird place and it's not as complete as your
implementation. I was just playing around with a toy problem at the time.

Cody

**Comment by SudiptaBiswas** [comment]
As mentioned by @permcody, I am removing the old `JouleHeating` kernel in this PR itself. This should close #8226 as well.

**Pull request merged**